test/sharness/t0170-dht.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/sh
+
+test_description="Test dht command"
+
+. lib/test-lib.sh
+
+# start iptb + wait for peering
+NUM_NODES=5
+test_expect_success 'init iptb' '
+  iptb init -n $NUM_NODES --bootstrap=none --port=0 &&
+  startup_cluster $NUM_NODES
+'
+
+test_expect_success 'peer ids' '
+  PEERID_0=$(ipfsi 0 id --format="<id>") &&
+  PEERID_2=$(ipfsi 2 id --format="<id>")
+'
+
+# ipfs dht findpeer <peerID>
+test_expect_success 'findpeer' '
+  ipfsi 1 dht findpeer $PEERID_0 >actual &&
+	egrep "/ip4/127.0.0.1/tcp/.*" actual >/dev/null &&
+	egrep "/ip4/.*/tcp/.*" actual >>/dev/null ||
+	test_fsh cat actual
+'
+
+# ipfs dht put <key> <value>
+test_expect_success 'put' '
+  ipfsi 1 dht put planet pluto >actual &&
+  PEERS=$(wc -l actual | cut -d '"'"' '"'"' -f 1) &&
+  [ -s actual ] ||
+	test_fsh cat actual
+'
+
+# ipfs dht findprovs <key>
+test_expect_success 'findprovs' '
+  ipfsi 4 dht findprovs planet >actual &&
+  PEERS=$(wc -l actual | cut -d '"'"' '"'"' -f 1) &&
```

Comment:
What is the correct way to do this? I've noticed that the output of `put` often does not match the output of `findprovs`. There seems to always be *an* intersection, but some keys are missing or are different. I'm not sure why this is.